src/VotingPlatform/Election/Listener/FinishVoteCommandListener.php

```diff
@@ -0,0 +1,118 @@
+use AppBundle\Repository\VotingPlatform\ElectionRepository;
+use AppBundle\Repository\VotingPlatform\VoterRepository;
+use AppBundle\VotingPlatform\Election\VoteCommand\VoteCommand;
+use AppBundle\VotingPlatform\Election\VoteCommandStateEnum;
+use Doctrine\ORM\EntityManagerInterface;
+use Symfony\Component\EventDispatcher\EventSubscriberInterface;
+use Symfony\Component\Security\Core\Security;
+use Symfony\Component\Workflow\Event\Event;
+
+class FinishVoteCommandListener implements EventSubscriberInterface
+{
+    private $entityManager;
+    private $security;
+    private $voterRepository;
+    private $electionRepository;
+    private $candidateGroupRepository;
+
+    public function __construct(
+        EntityManagerInterface $entityManager,
+        Security $security,
+        VoterRepository $voterRepository,
+        ElectionRepository $electionRepository,
+        CandidateGroupRepository $candidateGroupRepository
+    ) {
+        $this->entityManager = $entityManager;
+        $this->security = $security;
+        $this->voterRepository = $voterRepository;
+        $this->electionRepository = $electionRepository;
+        $this->candidateGroupRepository = $candidateGroupRepository;
+    }
+
+    public static function getSubscribedEvents()
+    {
+        return [
+            sprintf('workflow.voting_process.completed.%s', VoteCommandStateEnum::TO_FINISH) => ['persistVote'],
+        ];
+    }
+
+    public function persistVote(Event $event): void
+    {
+        $command = $event->getSubject();
+
+        if (!$command instanceof VoteCommand) {
+            return;
+        }
+
+        $election = $this->electionRepository->findByUuid($command->getElectionUuid());
+
+        if (!$election instanceof Election) {
+            return;
+        }
+
+        // 1. create vote history for the current voter
+        $vote = $this->generateVote($election);
+
+        // 2. generate a unique key to save the vote result with
+        $voterKey = $this->generateVoterKey();
+
+        // 3. create vote result with unique key
+        $voteResult = $this->createVoteResult($election, $command, $voterKey);
+
+        $this->entityManager->persist($vote);
+        $this->entityManager->persist($voteResult);
+
+        $this->entityManager->flush();
+    }
+
+    private function generateVote(Election $election): Vote
+    {
+        /** @var Adherent $adherent */
+        $adherent = $this->security->getUser();
+
+        if (!$voter = $this->voterRepository->findForAdherent($adherent)) {
+            $voter = new Voter($adherent);
+        }
+
+        return new Vote($voter, $election);
+    }
+
+    private function createVoteResult(Election $election, VoteCommand $command, string $voterKey): VoteResult
+    {
+        $voteResult = new VoteResult($election, $voterKey);
+
+        foreach ($command->getCandidateGroups() as $choice) {
+            $voteChoice = new VoteChoice();
+
+            if (VoteChoice::BLANK_VOTE_VALUE == $choice) {
```

Comment:
=== ?

Reply:
I have int type into const and string type coming from POST